config/routes.rb

```diff
@@ -3,6 +3,9 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+  resources :posts, only: [:new, :create]
+  resources :comments, only: [:new, :create]
```

Comment:
A comment is added to a specific post so the controller needs a post id. Kindly add new and create routes in such a way that they carry post id with them for comments controller.